ogre2/src/Ogre2BoundingBoxCamera.cc

```diff
@@ -0,0 +1,1469 @@
+      viewMatrix.extractQuaternion());
+    auto bodyWorldRotation = Ogre2Conversions::Convert(orientation);
+
+    // Body to camera rotation = body_world * world_camera
+    auto bodyCameraRotation = worldCameraRotation * bodyWorldRotation;
+    box->Orientation(bodyCameraRotation);
+
+    this->dataPtr->boundingboxes[ogreId] = box;
+    itor.moveNext();
+  }
+
+  // Set boxes labels
+  for (auto box : this->dataPtr->boundingboxes)
+  {
+    uint32_t ogreId = box.first;
+    uint32_t label = this->dataPtr->visibleBoxesLabel[ogreId];
+
+    box.second->Label(label);
+  }
+
+  // Combine boxes of multi-links model if exists
+  this->MergeMultiLinksModels3D();
+}
+
+/////////////////////////////////////////////////
+void Ogre2BoundingBoxCamera::VisibleBoundingBoxes()
+{
+  uint32_t width = this->ImageWidth();
+  uint32_t height = this->ImageHeight();
+  uint32_t channelCount = 3;
+
+  struct BoxBoundary
+  {
+    uint32_t minX;
+    uint32_t minY;
+    uint32_t maxX;
+    uint32_t maxY;
+  };
+
+  std::unordered_map<uint32_t, BoxBoundary*> boxesBoundary;
+
+  // find item's boundaries from panoptic BoundingBox
+  for (uint32_t y = 0; y < height; y++)
+  {
+    for (uint32_t x = 0; x < width; x++)
+    {
+      auto index = (y * width + x) * channelCount;
+
+      uint32_t label = this->dataPtr->buffer[index + 2];
+      uint32_t ogreId1 = this->dataPtr->buffer[index + 1];
+      uint32_t ogreId2 = this->dataPtr->buffer[index + 0];
+
+      if (label != this->dataPtr->materialSwitcher->backgroundLabel)
+      {
+        // get the OGRE id of 16 bit value
+        uint32_t ogreId = ogreId1 * 256 + ogreId2;
+
+        BoundingBox *box;
+        BoxBoundary *boundary;
+
+        // create new boxes when its first pixel appears
+        if (!this->dataPtr->boundingboxes.count(ogreId))
+        {
+          box = new BoundingBox();
+          box->Label(label);
+
+          boundary = new BoxBoundary();
+          boundary->minX = width;
+          boundary->minY = height;
+          boundary->maxX = 0;
+          boundary->maxY = 0;
+
+          boxesBoundary[ogreId] = boundary;
+          this->dataPtr->boundingboxes[ogreId] = box;
+        }
+        else
+        {
+          boundary = boxesBoundary[ogreId];
+        }
+
+        boundary->minX = std::min<uint32_t>(boundary->minX, x);
+        boundary->minY = std::min<uint32_t>(boundary->minY, y);
+        boundary->maxX = std::max<uint32_t>(boundary->maxX, x);
+        boundary->maxY = std::max<uint32_t>(boundary->maxY, y);
+      }
+    }
+  }
+
+  for (auto box : this->dataPtr->boundingboxes)
+  {
+    // Get the box's boundary
+    auto ogreId = box.first;
+    auto boundary = boxesBoundary[ogreId];
+    auto boxWidth = boundary->maxX - boundary->minX;
+    auto boxHeight = boundary->maxY - boundary->minY;
+
+    box.second->Center({boundary->minX + boxWidth / 2,
+        boundary->minY + boxHeight / 2, 0});
+    box.second->Size({boxWidth, boxHeight, 0});
```

Comment:
```suggestion
    box.second->Center({static_cast<double>(boundary->minX) + boxWidth / 2,
        static_cast<double>(boundary->minY) + boxHeight / 2, 0});

    box.second->Size({static_cast<double>(boxWidth),
        static_cast<double>(boxHeight), 0});
```

Prevent narrowing conversion warning.